cristin-import/src/main/java/no/unit/nva/cristin/lambda/CristinEntryEventConsumer.java

```diff
@@ -123,6 +127,41 @@ private static boolean isCristinAdditionalIdentifier(AdditionalIdentifier additi
         return CristinObject.IDENTIFIER_ORIGIN.equals(additionalIdentifier.getSource());
     }
 
+    private Publication persistPartOfCristinIdentiferIfPartOfExists(Publication publication,
```

Comment:
I am not pleased with this code. It is weird. First of all functions that returns an object without modifying it and instead changes stuff in the environment like a void function would. Secondly the multiple creation of cristinObject from the eventbody. 

I am open for suggestion on how to improve this. Right now I can't think of anything without refactor the processInput function to not use attempt, or functions that take in more than two parameters.